crates/sui-core/src/consensus_validator.rs

```diff
@@ -108,6 +114,13 @@ impl TransactionValidator for SuiTxValidator {
                     .wrap_err("Malformed batch (failed to verify)")
             })
             .await??;
+
+        // All checkpoint sigs have been verified, forward them to the checkpoint service
+        for ckpt in ckpt_messages {
+            self.checkpoint_service
+                .notify_checkpoint_signature(&self.epoch_store, &ckpt)?;
```

Comment:
Unlike in consensus_handler, we have not verified that the sender matches the name in the checkpoint sig. Is this important? If so, I need some advice on how to find the originator of the batch here.

Reply:
To be honest it don't seem _that_ important to me - after all it's a valid signature, sure it's suspicious/odd that a node is distributing someone else's signature but I don't see a security issue or something. The batch it self doesn't contain the originator, so we'll need to inject that , off the top of my head, via two places:
* when it gets broadcasted by the source
* when is requested from the source

but it might not worth the effort

Reply:
Yeah, I think since we checked signature validity, it does not matter who sent it